### PR TITLE
feat(browser): add support for browser-request in request class, resolves #234

### LINKED ISSUE #234: Support `fetch` as an `http` backend for the swagger client
The swagger client currently depends on a [request](https://www.npmjs.com/package/request)-based [HTTP request object](https://github.com/godaddy/kubernetes-client/blob/master/lib/request.js) for making HTTP calls.  This is a blocker for running kubernetes-client on browsers.

Implement a `fetch`-based HTTP request object.

### PR DESCRIPTION
**Pull request by cfellin1**

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
-const request = require('request');
+// If in browser, use request-browser; else use normal request.
+const request = Window ? require('request-browser') : require('request');
```
Comment (silasbw) [inline lib/request.js]: `typeof Window !== 'undefined' ? ...`

Also, where does request-browser come from?  do we need to update package.json and package-lock.json?

**Comment by xmlking** [comment]
can we use  to `fetch` for browser and use `node-fetch` as polyfill  for node ?

**Comment by cfellin1** [comment]
@xmlking What's the benefit to justify changing the request wrapper class?

**Comment by xmlking** [comment]
Fetch is natively promise. No polyfills required for browsers.

**Comment by jeffpearce** [comment]
So it's polyfill for node vs polyfill for browser? Did I understand correctly? What are the tradeoffs of one vs the other?

**Comment by xmlking** [comment]
Yes. Backend polyfill can be removed when node supports it. Keeping browser more stable is important in my opinion

**Comment by cfellin1** [comment]
Closing this PR: if we go with the fetch route, the end result will look pretty different than this.

**Pull request closed without merging**